libsignal-service/src/push_service.rs

```diff
@@ -720,9 +718,9 @@ pub trait PushService: MaybeSend {
         device_id: u32,
     ) -> Result<Vec<PreKeyBundle>, ServiceError> {
         let path = if device_id == 1 {
-            format!("/v2/keys/{}/*", destination.identifier())
+            format!("/v2/keys/{}/*", destination)
         } else {
-            format!("/v2/keys/{}/{}", destination.identifier(), device_id)
+            format!("/v2/keys/{}/{}", destination, device_id)
```

Comment:
I would explicitely ask the destination.uuid here. Display is for user-friendly representation, not necessarily machine-useful representation. See below.

Reply:
I have remove the `impl fmt::Display` altogether, as I think it can be misused and we don't really need to ever print the `ServiceAddress` as-is.